src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/RandomNumberGenerator.cs

```diff
@@ -24,20 +20,30 @@ public void Dispose()
             return;
         }
 
-        protected virtual void Dispose(bool disposing)
-        {
-            return;
-        }
+        protected virtual void Dispose(bool disposing) { }
 
         public abstract void GetBytes(byte[] data);
 
-        internal void ValidateGetBytesArgs(byte[] data)
+        public virtual void GetBytes(byte[] data, int offset, int count)
         {
-            if (data == null)
+            if (data == null) throw new ArgumentNullException(nameof(data));
+            if (offset < 0) throw new ArgumentOutOfRangeException(nameof(offset), SR.ArgumentOutOfRange_NeedNonNegNum);
+            if (count < 0) throw new ArgumentOutOfRangeException(nameof(count), SR.ArgumentOutOfRange_NeedNonNegNum);
+            if (offset + count > data.Length) throw new ArgumentException(SR.Argument_InvalidOffLen);
+
+            if (count > 0)
             {
-                throw new ArgumentNullException(nameof(data));
+                byte[] tempData = new byte[count];
+                GetBytes(tempData);
+                Array.Copy(tempData, 0, data, offset, count);
```

Comment:
Ouch.  Per my previous comment, we should change the interop signatures to accept ```byte*``` and put the primary logic in ```GetBytes(byte[] data, int offset, int count)``` rather than in ```GetBytes(byte[] data)```, so that we don't have to allocate and copy here.